diff --git a/CMakeLists.txt b/CMakeLists.txt
index dddbe1fa9..11df81324 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -722,10 +722,11 @@ endif()
 ### Eigen (optional) #############################
 if(ENABLE_EIGEN3)
     find_package(Eigen3 QUIET)
-    if(EIGEN3_FOUND)
-	    message(STATUS "Found Eigen3 library: ${EIGEN3_INCLUDE_DIR} (version \"${EIGEN3_VERSION_STRING}\")")
+    if(Eigen3_FOUND)
+        get_target_property(EIGEN3_INCLUDE_DIR Eigen3::Eigen INTERFACE_INCLUDE_DIRECTORIES)
+        message(STATUS "Found Eigen3 library: ${EIGEN3_INCLUDE_DIR} (version \"${Eigen3_VERSION}\")")
         add_definitions(-DHAVE_EIGEN3)
-	add_definitions(-DEIGEN3_VERSION_STRING=\"${EIGEN3_VERSION_STRING}\")
+	add_definitions(-DEIGEN3_VERSION_STRING=\"${Eigen3_VERSION}\")
         include_directories(${EIGEN3_INCLUDE_DIR})
     else()
         message(STATUS "Eigen3 library NOT FOUND.")
